src/Storages/MeiliSearch/SinkMeiliSearch.cpp

```diff
@@ -0,0 +1,106 @@
+#include <Storages/MeiliSearch/SinkMeiliSearch.h>
+#include "Core/Field.h"
+#include "base/JSON.h"
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int MEILISEARCH_EXCEPTION;
+}
+
+SinkMeiliSearch::SinkMeiliSearch(
+    const MeiliSearchConfiguration& config_,
+    const Block& sample_block_,
+    UInt64 max_block_size_)
+    : SinkToStorage(sample_block_)
+    , connection(config_)
+    , max_block_size{max_block_size_}
+    , sample_block{sample_block_} {}
+
+String quotify(const String& s);
+
+String getStringRepresentation(const ColumnWithTypeAndName& col, size_t row) {
+    Field elem;
+    if (col.column->size() <= row) {
+        return "";
+    }
+    col.column->get(row, elem);
+    if (elem.getType() == Field::Types::Int64) {
+        return std::to_string(elem.get<Int64>());
+    } else if (elem.getType() == Field::Types::UInt64) {
+        return std::to_string(elem.get<UInt64>());
+    } else if (elem.getType() == Field::Types::String) {
+        return quotify(elem.get<String>());
+    }
+    return "";
+}
+
+String SinkMeiliSearch::getOneElement(const Block& block, int ind) {
+    String ans = "{";
+    int id = 0;
+    for (const auto& col : block) {
+        ans += quotify(sample_block.getByPosition(id++).name) + ":" + getStringRepresentation(col, ind) + ",";
+    }
+    ans.back() = '}';
```

Comment:
No need to manually convert Block to json. You can just use `JSONRowOutputFormat` similar to how it is done here:
https://github.com/ClickHouse/ClickHouse/blob/master/programs/odbc-bridge/ODBCBlockOutputStream.cpp#L37-L39

Reply:
As far as I understand, JSONRowOutputFormat creates json with all String fields, but I would like to support the same field types in both ClickHouse and MeiliSearch